src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioTransformationServiceImpl.java

```diff
@@ -60,4 +71,31 @@ public List<MediaEntity> getMediaMetadata(Integer hearingId) {
         return mediaRepository.findAllByHearingId(hearingId);
     }
 
+    @Override
+    public Optional<UUID> getMediaLocation(MediaEntity media) {
+        Optional<UUID> externalLocation = Optional.empty();
+
+        ObjectDirectoryStatusEnum statusEnum = ObjectDirectoryStatusEnum.STORED;
+        ObjectDirectoryStatusEntity objectDirectoryStatus = objectDirectoryStatusRepository.getReferenceById(statusEnum.getId());
+        String externalLocationType = "unstructured";
+        List<ExternalObjectDirectoryEntity> externalObjectDirectoryEntityList = externalObjectDirectoryRepository.findByMediaStatusAndType(
+            media, objectDirectoryStatus, externalLocationType
+        );
+
+        if (!externalObjectDirectoryEntityList.isEmpty()) {
+            if (externalObjectDirectoryEntityList.size() != ONE) {
+                log.warn(
```

Comment:
We would need it throw an exception but it can be handled in subsequent ticket. addAudio adds the metadata into moj_media table and insert into eod table as INBOUD type. Then an automated job moves the files around and update the EOD table with corresponding status. if the automated job fails to move data for any reason, EOD will not have a record for that media.





[5:56](https://hmcts-reform.slack.com/archives/D053QR2QHKR/p1688489818807049)
then an automated job moves the files around and update the EOD table with corresponding status.
[5:57](https://hmcts-reform.slack.com/archives/D053QR2QHKR/p1688489843955869)
if the automated job fails to move data for any reason, EOD will not have a record for that media.